<?xml version="1.0"?>
<robot name="ur10_robot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Include the UR10 robot description -->
  <xacro:include filename="$(find nrs_vision_rviz)/urdf/ur10.xacro" />
  <!-- Define the testbed link -->
  <link name="testbed">
    <visual>
      <origin xyz="0.3 0.5 -0.45" rpy="0 0 0" />
      <geometry>
        <box size="0.84 1.24 0.9"/> <!-- 840mm x 1240mm x 900mm -->
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <origin xyz="0.3 0.5 -0.45" rpy="0 0 0" />
      <geometry>
        <box size="0.84 1.24 0.9"/> <!-- 840mm x 1240mm x 900mm -->
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </collision>
    <inertial>
      <mass value="100" />
      <origin xyz="0.3 0.5 -0.45" rpy="0 0 0" />
      <inertia ixx="33.33" ixy="0.0" ixz="0.0" iyy="33.33" iyz="0.0" izz="33.33" />
    </inertial>
  </link>


  <link name="tooltip">
    <visual>
      <geometry>
        <mesh filename="package://nrs_vision_rviz/mesh/ur10_tooltip.stl"/>
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://nrs_vision_rviz/mesh/ur10_tooltip.stl"/>
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </collision>
  </link>

  <link name="workpiece">
    <visual>
      <geometry>
        <mesh filename="package://nrs_vision_rviz/mesh/fenda_wrap.stl"/>
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <!-- <collision>
      <geometry>
        <mesh filename="package://nrs_vision_rviz/mesh/sample_wrap.stl"/>
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </collision> -->
  </link>



  <!-- Define a fixed joint to attach the testbed to the base link -->
  <joint name="testbed_joint" type="fixed">
    <parent link="base_link" />
    <child link="testbed" />
    <origin xyz="0 0 0" rpy="0 0 -0.7853981633974483" /> <!-- 45 degrees in radians -->
  </joint>

  <!-- Define a fixed joint to attach the testbed to the base link -->
  <joint name="tooltip_joint" type="fixed">
    <parent link="wrist_3_link" />
    <child link="tooltip" />
    <origin xyz="0 0 0" rpy="0 0 -1.5707963267948966" /> 
  </joint>

  <!-- Define a fixed joint to attach the workpiece to the base link -->
  <joint name="workpiece_joint" type="fixed">
    <parent link="base_link" />
    <child link="workpiece" />
    <origin xyz="0.0 0.0 0.0" rpy="0 0 0" /> 
  </joint>

</robot>